q2_aldex2/_method.py

```diff
@@ -48,6 +61,32 @@ def aldex2(table: pd.DataFrame,
                             " and stderr to learn more." % e.returncode)
 
         summary = pd.read_csv(summary_fp, index_col=0)
-        differentials = summary[['effect']]
-        # don't return summary for now (TODO!)
-        return differentials
+        #differentials = summary[['effect']]
+	# hack to fix column name for features because aldex removes
+	#it in R because of row.names = 1
+
+        summary.index.name = "featureid"
+        return summary
+
+def extract_differences(table: pd.DataFrame, sig_threshold: float = 0.1, effect_threshold: float = 1, difference_threshold: float = 1, test: str = 'welch') -> pd.DataFrame:
+
+    # checks to make sure there is no error
+    # ensure max or min, depending on case
+
+    effect_statistic_function = _effect_statistic_functions[test]
+
+    if sig_threshold < table[effect_statistic_function].min():
+        raise ValueError("You have selected a significance threshold that is lower than minimum Q score (-p--sig-threshold). Select a higher threshold.")
+
+    # absolute values needed for effect or difference to see change in either
+    # condition
+    if effect_threshold > abs(table['effect']).max():
+        raise ValueError("You have selected an effect threshold that exceeds maximum effect size (-p--effect-threshold). Choose a lower threshold, or be aware that there there will be no features in the output.")
+
+    if difference_threshold > abs(table['diff.btw']).max():
+        raise ValueError("You have selected a difference threshold that exceeds maximum difference (-p--difference-threshold). Choose a lower threshold, or be aware that there will be no features in the output.")
+
+    # subset the table if it psases all the threshold
+    differentials_sig = table[(table[effect_statistic_function] <= sig_threshold) & (abs(table['effect']) > effect_threshold) & (abs(table['diff.btw']) > difference_threshold)]
```

Comment:
Totally optional - but a good python practice is to have everything within 80 characters.  
So lines like the above line will be split across 2-3 lines.  It is easier for future developers to read the code.

But I'll leave it up to you if you want to do this or not.

Reply:
fixed!